File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        // Using Hashing
        // int n = nums.size();
        // vector<int>hash(n+1,0);
        // for(int i =0; i<n; i++){
        //     hash[nums[i]]= 1;
        // }
        // // covers the entire loop
        // for(int i =0;i<=n;i++){
        //     if(hash[i] == 0){
        //         return i;
        //     }
        // }
        // return -1;

        // Using sum method
        int n = nums.size();
        int sum = n*(n+1)/2;
        int s1 = 0;
        for(int i = 0; i<n; i++){
            s1 += nums[i];
        }
        return sum - s1;
    }
};

File: 268-missing-number/missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        # for i in range(0, len(nums)+1):
        #     if i not in nums:
        #         return i

        freq = {}

        for i in range(0,len(nums)+1):
            freq[i] = 0

        for num in nums:
            freq[num] = 1
        
        for key, value in freq.items():
            if value == 0:
                return key